tvnamer: fix building on all platforms

knl opened the pull request:
###### Description of changes

This change unblocks tvnamer by using the latest version of tvnamer. In addition, it fixes the build of requests-cache 0.5.2 dependency, that was missing setuptools.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

mweinelt commented on pkgs/tools/misc/tvnamer/default.nix:
```diff
@@ -14,6 +15,10 @@ let
           sha256 = "sha256-gTAjJpaGBF+OAeIonMHn6a5asi3dHihJqQk6s6tycOs=";
         };
 
+        nativeBuildInputs = with python3Packages; [
```
```suggestion
        nativeBuildInputs = with python3.pkgs; [
```

knl replied:
I couldn't use `super` here, as it's super coming from `overridePythonAttrs`. I did a bit of renaming, hope it's better now.

mweinelt commented on pkgs/tools/misc/tvnamer/default.nix:
```diff
@@ -1,5 +1,6 @@
 { lib
 , python3
+, python3Packages
```
```suggestion
```

knl replied:
Thanks, I used this change and is part of the new version.

mweinelt commented on pkgs/tools/misc/tvnamer/default.nix:
```diff
@@ -4,16 +4,20 @@
+    packageOverrides = final: prev: rec {
       # tvdb_api v3.1.0 has a hard requirement on requests-cache < 0.6
-      requests-cache = super.requests-cache.overridePythonAttrs (super: rec {
+      requests-cache = prev.requests-cache.overridePythonAttrs (o: rec {
```
```suggestion
      requests-cache = prev.requests-cache.overridePythonAttrs (oldAttrs: rec {
```

knl replied:
Applied and had to change one more instance. This is way more clear, thanks!

mweinelt commented:
And the update to the python library should be in its own commit.

Commit:
tvdb_api: unblock by using the latest version

Unreleased version (there have not been any updates since April 2021)
contains fixes to _to_bytes issue that caused tvdb_api to fail to build.

Commit:
tvnamer: add setuptools dependency to requests-cache 0.5.2

tvnamer has a direct dependency on old version of requests-cache 0.5.2.
Unfortunately, it was failing to build as it was missing setuptools.
This change adds this dependency. Along with the fixes to tvdb_api, now
tvnamer correctly builds.

peterhoeg commented:
Nice, thanks!
